app/api/dao/user.py

```diff
@@ -235,77 +235,33 @@ def update_user_profile(user_id: int, data: Dict[str, str]):
                     messages.USER_USES_A_USERNAME_THAT_ALREADY_EXISTS,
                     HTTPStatus.BAD_REQUEST,
                 )
-
             user.username = username
 
-        if "name" in data and data["name"]:
-            user.name = data["name"]
+        user.name = data.get("name", None)
```

Comment:
If you use data["name"], then KeyError will be raised, not with data.get
```suggestion
        user.name = data.get("name")
```